packages/block-editor/src/components/block-navigation/block-select-button.js

```diff
@@ -61,6 +62,7 @@ function BlockNavigationBlockSelectButton(
 				onDragEnd={ onDragEnd }
 				draggable={ draggable }
 			>
+				<BlockNavigationExpander onClick={ onClick } />
```

Comment:
Wonder if we should separate `onClick` with a new prop like `onToggleExpanded`.

`onClick` implies listening for the click on the entire component's surface, which... is the case but we prevent triggering the toggle in `toggleExpandOrSelectBlock` with the `forceToggle` parameter.

It looks a bit convoluted and unclear to me, and I think that two separate props, one for selecting the block and one for toggling the branch, would make the component easier to consume. 🤔

Reply:
Thanks, I'll see if I can make it clearer. I think I might have left it that way from experimenting with a few different state approaches.

Reply:
I tried teasing those two functions out in https://github.com/WordPress/gutenberg/pull/32117/commits/1b8b13d37b1f1306d9291e0a1cd6f8dfbb4ee321. There's a bit of prop passing, but let me know what you think.